### DISCUSSION TITLE
Support for ICND1063 LED driver chip

### DISCUSSION THREAD

**Discussion by PodmogilnyiEvgenii** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#630] [General]
Hi!
Do you have any plans to support ICND1063 chips?

Some photos:
![WhatsApp Image 2024-04-27 at 20 37 26 (1)](https://github.com/mrcodetastic/ESP32-HUB75-MatrixPanel-DMA/assets/102897031/2d62b822-2164-4afa-a418-66386a6fffb7)
![WhatsApp Image 2024-04-27 at 20 37 27 (1)](https://github.com/mrcodetastic/ESP32-HUB75-MatrixPanel-DMA/assets/102897031/6f8040c5-7fdc-4628-aa27-e30b70140ed0)
![WhatsApp Image 2024-04-27 at 20 37 27 (2)](https://github.com/mrcodetastic/ESP32-HUB75-MatrixPanel-DMA/assets/102897031/dff7fa2e-45df-4b41-bd85-7cf8919f1daf)
![WhatsApp Image 2024-04-27 at 20 37 27](https://github.com/mrcodetastic/ESP32-HUB75-MatrixPanel-DMA/assets/102897031/1665223f-6151-4b99-aa43-b107c3315069)

**Comment by board707** [comment]
Do you have a datasheet for it?

**Reply by PodmogilnyiEvgenii** [reply]
No, but I'll try to find it.

**Reply by PatrickStar8888** [reply]
I have datasheet of ICND1065, but 1063 is really different from 1065. If you want to driver this LED board, you can try Colorlight E120 or Monncell.

**Reply by board707** [reply]
> I have datasheet of ICND1065,

Could you share it?

**Reply by PatrickStar8888** [reply]
Definitely Yes, tell me your email address. 
In fact, I am Chinese, and the datasheet is also in Chinese, If you need translation just tell me

**Reply by board707** [reply]
Please send the file to bort707h@gmail.com
Thanks

**Reply by PatrickStar8888** [reply]
Bro I have sent

**Reply by Muthyam-D** [reply]
@PatrickStar8888 Do you have datasheet of ICND1012AP?

**Reply by Exceed-I** [reply]
> I have datasheet of ICND1065, but 1063 is really different from 1065. If you want to driver this LED board, you can try Colorlight E120 or Monncell.
Hi @PatrickStar8888, Could you share the datasheet of ICND1065? Thanks!

**Reply by PatrickStar8888** [reply]
> > I have datasheet of ICND1065, but 1063 is really different from 1065. If you want to driver this LED board, you can try Colorlight E120 or Monncell.
> > Hi @PatrickStar8888, Could you share the datasheet of ICND1065? Thanks!

The datasheet is in Chinese, check the pdf~

[ICND1065_datasheet_CN_2022_V1.0.pdf](https://github.com/user-attachments/files/22458678/ICND1065_datasheet_CN_2022_V1.0.pdf)

**Reply by PatrickStar8888** [reply]
> @PatrickStar8888 Do you have datasheet of ICND1012AP?

Sorry, I don't have. But I know ICND2012 is a 8-channel source driver, and work like a 138 decoder. So it is easy to control.

**Reply by board707** [reply]
Hi @Muthyam-D 
As @PatrickStar8888  said, the ICDN2012 is a analog of 74HC138. The datashhet is very short
https://datasheet4u.com/pdf-down/I/C/N/ICN2012-CHIPONE.pdf

**Reply by Exceed-I** [reply]
> > > I have datasheet of ICND1065, but 1063 is really different from 1065. If you want to driver this LED board, you can try Colorlight E120 or Monncell.
> > > Hi @PatrickStar8888, Could you share the datasheet of ICND1065? Thanks!
> 
> The datasheet is in Chinese, check the pdf~
> 
> [ICND1065_datasheet_CN_2022_V1.0.pdf](https://github.com/user-attachments/files/22458678/ICND1065_datasheet_CN_2022_V1.0.pdf)

Thank you so much~

**Comment by board707** [comment]
Are you sure that you need a ICN1012 and not a 2012 chip?
I didn't heard about a ICN1012 driver  before.

**Reply by Muthyam-D** [reply]
Yeah, really I'm talking about ICND1012AP which is a sourcing IC not a sinking one.

**Comment by rigorka** [comment]
Hi @board707, am I right assuming that your code for S-PWM FM6353 is not ported to this ESP32 library, and is only available for STM32?
The reason for this question is that I've got some S-PWM ICND1065 panels handy, and was curious to check if I could get them running using your FM6353 driver approach, same as you did with ICND2153.

**Reply by board707** [reply]
Hi @rigorka 

> am I right assuming that your code for S-PWM FM6353 is not ported to this ESP32 library?

Yes, that's right, my code is written for STM32 and I didn't port it to ESP32. However, there is another project for this library, written for ICN2053 drivers (see #324)

However, I don't really understand how this will help you with  ICND1065 panels . Are You think that your chips are analogs of FM6353?

**Reply by rigorka** [reply]
> Yes, that's right, my code is written for STM32 and I didn't port it to ESP32. However, there is another project for this library, written for ICN2053 drivers (see #324)

Thanks for pointing me to LAutor's fork of this lib, I have somehow missed it. Will certainly take a look.
 
> However, I don't really understand how this will help you with ICND1065 panels . Are You think that your chips are analogs of FM6353?

I don't know at the moment, but would like to check just in case they are in fact similar. I was not able to find the datasheet for ICND1065 so it's a bit of a wild guess.  I'm not getting my hopes up too high, but since I already have these panels just sitting here, might as well give it a shot.

**Comment by patkepa** [comment]
Did anyone manage any progress on the ICND1065 chip?

**Reply by board707** [reply]
> Did anyone manage any progress on the ICND1065 chip?

The [DMD_STM32](https://github.com/board707/DMD_STM32) library README declares support for this chip (along with about a dozen others sPWM drivers).

**Reply by patkepa** [reply]
Thank you very much! I'll check it out.
